src/crystal/main.cr

```diff
@@ -101,6 +110,20 @@ module Crystal
   def self.main_user_code(argc : Int32, argv : UInt8**)
     _crystal_main(argc, argv)
   end
+
+  # :nodoc:
+  def self.remember_blocking_state
+    @@stdin_is_blocking = IO::FileDescriptor.fcntl(0, LibC::F_GETFL) & LibC::O_NONBLOCK == 0
```

Comment:
Don't we have `blocking?`

Reply:
You mean doing `STDIN.blocking`? At that point `STDIN` blocking state would have changed already.

Reply:
(though we can change the logic in the constructor and do it in Crystal.main, I think it's a good idea)